cmd/vulcan-nuclei/main.go

```diff
@@ -0,0 +1,269 @@
+	tf, err := ioutil.ReadDir(nucleiTemplatePath)
+	if err != nil {
+		return errors.New("unable to read nuclei templates folder")
+	}
+
+	for _, f := range tf {
+		if f.IsDir() {
+			availableTemplates[f.Name()] = void{}
+		}
+	}
+	logger.Infof("Available templates: %+v", availableTemplates)
+
+	selectedTemplates := make(map[string]void)
+	inclusionList := strings.Split(opt.TemplateInclusionList, ",")
+	inclusionList = append(inclusionList, defaultInclusionList...)
+	// If inclusionList is not empty, use only inclued ones.
+	// If inclusionList is empty, user all available templates.
+	logger.Infof("Included templates: %+v", inclusionList)
+	if len(inclusionList) > 1 {
+		for _, v := range inclusionList {
+			if _, ok := availableTemplates[v]; ok {
+				selectedTemplates[v] = void{}
+			}
+		}
+	} else {
+		for k := range availableTemplates {
+			selectedTemplates[k] = void{}
+		}
+	}
+
+	// Remove exclued templates from selectedTemplates.
+	exclusionList := strings.Split(opt.TemplateExclusionList, ",")
+	exclusionList = append(exclusionList, defaultExclusionList...)
+	logger.Infof("Exclued templates: %+v", exclusionList)
+	for _, v := range exclusionList {
+		if _, ok := selectedTemplates[v]; ok {
+			delete(selectedTemplates, v)
+		}
+	}
+
+	logger.Infof("Selected templates: %+v", selectedTemplates)
+	// No templates selected. Return.
+	if len(selectedTemplates) < 1 {
+		return nil
+	}
+
+	// Build arguments.
+	nucleiArgs := []string{
+		"-target", target,
+		"-c", "20",
+		"-json",
+		"-silent",
+		"-H", userAgent,
+	}
+
+	// Include only selected severities.
+	if opt.Severities != "" {
+		logger.Infof("Selected severities: %s", opt.Severities)
+		severities := []string{"-severity", opt.Severities}
+		nucleiArgs = append(nucleiArgs, severities...)
+	}
+
+	logger.Debugf("nuclei command args: [%s]", nucleiArgs)
+	var vulns []Vulnerability
+	for t := range selectedTemplates {
+		nucleiArgsWithTemplate := append(nucleiArgs, "-t", t)
+		output, err := runNucleiCmd(nucleiArgsWithTemplate)
+		if err != nil {
+			logger.Errorf("nuclei execution failed with template [%s]: %s", t, err)
+			continue
+		}
+		if len(output) == 0 {
+			logger.Infof("no vulnerabilities found with template [%s]", t)
+			continue
+		}
+		scanner := bufio.NewScanner(bytes.NewReader(output))
+		for scanner.Scan() {
+			var v Vulnerability
+			err := json.Unmarshal(scanner.Bytes(), &v)
+			if err != nil {
+				logger.Errorf("unable to unmarshal vulnerability [%s]: %s", scanner.Text(), err)
+				continue
+			}
+			vulns = append(vulns, v)
+		}
+	}
+	// No vulnerabilities found. Return.
+	if len(vulns) < 1 {
+		logger.Info("no vulnerabilities found")
+		return nil
+	}
+
+	logger.Infof("nuclei found [%d] vulnerabilities", len(vulns))
+	for _, v := range vulns {
+		score := getScore(v.Severity)
+		var vuln = report.Vulnerability{
+			Summary:     v.Name,
+			Description: v.Description,
+			Recommendations: []string{
+				"Update or protect the exposed resource.",
```

Comment:
I see that Nuclei provides no recommendations, which is unfortunate. However, since the vulnerabilities detected are so varied I would just rather not report any recommendation at all than this generic line. The reason is that I feel users should see real value in the recommendations and that this recommendation does not apply to some vulnerabilities such as "WAF Detection", "Wappalyzer Technology Detection" and most likely many others.

Reply:
Agree. It doesn't add any value to the user/output.
Removed.